### PR TITLE
adds the argo workflow to run terraform templates

### PR DESCRIPTION
**Pull request by nimakaviani**
creates a new example runner for launching terraform templates from within the idpBuilder

### CODE REVIEW

File: examples/terraform-integrations/terraform-workflows-templates.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  namespace: argocd
+  labels:
+    env: dev
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/cnoe-io/backstage-terraform-integrations
+    targetRevision: main
```
Comment (nabuskey) [inline examples/terraform-integrations/terraform-workflows-templates.yaml:14]: Do we want to pin to a version?

Comment (nimakaviani) [reply examples/terraform-integrations/terraform-workflows-templates.yaml:14]: good call. I'd say at a later point in time. Right now this whole work is experimental and for demo purposes.

**Comment by nimakaviani** [comment]
cc @elamaran11

**Review by nabuskey — COMMENTED** [review]
Let's add a readme and let people know it's still under construction

**Comment by nimakaviani** [comment]
sounds good, will make the change.

**Review by nabuskey — APPROVED** [review]
LGTM

**Comment by csantanapr** [comment]
@nimakaviani Is this missing the backstage templates?

**Comment by elamaran11** [comment]
@csantanapr Backstage templates are in [this](https://github.com/cnoe-io/backstage-terraform-integrations) repo and should be added to the `backstage-config` and documentation is all in readme.

**Pull request merged**